packages/dai-plugin-mcd/src/SavingsService.js

```diff
@@ -107,6 +107,24 @@ export default class SavingsService extends PublicService {
     return getDsrEventHistory(this, address, this._eventHistoryCache);
   }
 
+  async getEarningsToDate(address) {
+    if (!this._eventHistoryCache) this._eventHistoryCache = {};
+    const eventHistory = await getDsrEventHistory(
+      this,
+      address,
+      this._eventHistoryCache
+    );
+    let sum = new BigNumber(0);
+    eventHistory.forEach(({ type, amount }) => {
+      if (type === 'DSR_DEPOSIT') sum = sum.plus(amount);
+      if (type === 'DSR_WITHDRAW') sum = sum.minus(amount);
+    });
+    // must cast to BN or this will throw if creating a negative currency amount
+    const balance = (await this.balance()).toBigNumber();
```

Comment:
@levity does `balanceOf(guy)` factor in the chi during its calculation? It looks like it should, but in my test I'm seeing a result that is less than the deposit amount. If not, then I would probably need to multiply the balance by chi on this line here.

Reply:
i'm not seeing any issues when running this locally... ?

Reply:
If I console log the balance when running the `'earnings to date'` test, the balance is `9.99999999999999999981`, though the amount joined was `10`.

Reply:
How did you console.log it? That value means Javascript rounding is happening (e.g. try `9.99999999999999999981 === 10`).

Reply:
I was logging `console.log(balance.toBigNumber()` in the `getEarningsToDate` function under test. I see that `9.99999999999999999981 === 10` is true, but `balance.eq(10)` is actually false. That's what's throwing me off.

Reply:
if the actual BigNumber value is 9.99999999999999999981, that means the rounding error happened before you created the BigNumber. if you can consistently reproduce this then there's a big somewhere in the Javascript.

Reply:
I realized that the numbers we're looking at here represent sub-wei amounts, which are smaller than the Dai adapter can handle anyway. So I added some code to round to the nearest wei. I'm not totally sure this is the right way to go--depends on the resolution of the issue in the smart contract, I guess. Take a look at my changes and let me know what you think